File: python_practice/bubble-sort-practice/bubble_sort.py
# An optimized version of Bubble Sort 
def bubble_sort(arr): 
    n = len(arr) 
    for i in range(n): 
        swapped = False
        for j in range(0, n-i-1): 
            if arr[j] > arr[j+1] : 
                temp = arr[j] 
                arr[j] = arr[j+1]
                arr[j+1] = temp 
                swapped = True
        if swapped == False: 
            break
    return arr





print(bubble_sort([3,2,1,4,5,1025,3]))
print(bubble_sort([1,1,1,1,2,4,9,5,555,3,1,1]))
print(bubble_sort([0,55,1,3,99,15,0,0]))

File: python_practice/bubble-sort-practice/bubble_sort.js
const bubbleSort =(arr)=>{
    l = arr.length
    for(i=0; i<l; i++){
        sort = false
        for(j=0; j<l-i-1; j++){
            if(arr[j] > arr[j+1]){
                temp = arr[j]
                arr[j] = arr[j+1]
                arr[j+1] = temp
                sort = true
            }
        }
        if(sort == false){
            break
        }
    }
    return arr
}

console.log(bubbleSort([3,2,1,4,5,1025,3]))
console.log(bubbleSort([1,1,1,1,2,4,9,5,555,3,1,1]))
console.log(bubbleSort([0,55,1,3,99,15,0,0]))